packages/svelte/src/performance.ts

```diff
@@ -0,0 +1,93 @@
+import { getCurrentHub } from '@sentry/browser';
+import { Span, Transaction } from '@sentry/types';
+import { afterUpdate, beforeUpdate, onMount } from 'svelte';
+import { current_component } from 'svelte/internal';
+
+import { DEFAULT_COMPONENT_NAME, UI_SVELTE_MOUNT, UI_SVELTE_UPDATE } from './constants';
+import { TrackComponentOptions } from './types';
+
+const defaultOptions: Required<Pick<TrackComponentOptions, 'trackMount' | 'trackUpdates'>> &
+  Pick<TrackComponentOptions, 'componentName'> = {
+  trackMount: true,
+  trackUpdates: true,
+};
+
+/**
+ * Tracks the Svelte component's intialization and mounting operation as well as
+ * updates and records them as spans.
+ * This function is injected automatically into your Svelte components' code
+ * if you are using the Sentry componentTrackingPreprocessor.
+ * Alternatively, you can call it yourself if you don't want to use the preprocessor.
+ */
+export function trackComponent(options?: TrackComponentOptions): void {
+  const mergedOptions = { ...defaultOptions, ...options };
+
+  const transaction = getActiveTransaction();
+  if (!transaction) {
+    return;
+  }
+
+  const customComponentName = mergedOptions.componentName;
+
+  // current_component.ctor.name is likely to give us the component's name automatically
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const componentName = `<${customComponentName || current_component.constructor.name || DEFAULT_COMPONENT_NAME}>`;
+
+  let mountSpan: Span | undefined = undefined;
+  if (mergedOptions.trackMount) {
+    mountSpan = recordMountSpan(transaction, componentName);
+  }
+
+  if (mergedOptions.trackUpdates) {
+    recordUpdateSpans(componentName, mountSpan);
+  }
+}
+
+function recordMountSpan(transaction: Transaction, componentName: string): Span {
+  const mountSpan = transaction.startChild({
+    op: UI_SVELTE_MOUNT,
+    description: componentName,
+  });
+
+  onMount(() => {
+    mountSpan.finish();
+  });
+
+  return mountSpan;
+}
+
+function recordUpdateSpans(componentName: string, mountSpan?: Span): void {
+  let updateSpan: Span | undefined;
+  beforeUpdate(() => {
+    // We need to get the active transaction again because the initial one could
+    // already be finished or there is currently no transaction going on.
+    const transaction = getActiveTransaction();
+    if (!transaction) {
+      return;
+    }
+
+    // If we are mounting the component when the update span is started, we start it as child
+    // of the mount span. Else, we start it as a child of the transaction.
```

Comment:
Could you elaborate a bit why the update span is a child of the component's mount span? Without having thought about it too much it doesn't click for me but maybe there's a reason.

Reply:
So, tbh this is totally up for discussion but the reason why I think it makes sense to make the update a child span of the mount span is because in the component lifecycle of Svelte, the `beforeUpdate` hook will be called before the mounting of the component is finished. Meaning, the first update of a component is part of its initial initialization. So overall, I think it makes sense to let this update (and only this one) be a child of the mount span.

Reply:
Ah ok. The way you explained this makes a lot of sense. I'd keep it too then!